android/beagle/src/main/java/br/com/zup/beagle/android/context/operations/exception/ExceptionController.kt

```diff
@@ -0,0 +1,185 @@
+ * limitations under the License.
+ */
+
+package br.com.zup.beagle.android.context.operations.exception
+
+import br.com.zup.beagle.android.context.operations.exception.strategy.ExceptionOperationTypes
+import br.com.zup.beagle.android.context.operations.exception.strategy.ExceptionParameterTypes
+import br.com.zup.beagle.android.context.operations.grammar.GrammarChars
+import br.com.zup.beagle.android.context.operations.operation.Operation
+import br.com.zup.beagle.android.context.operations.strategy.Operations
+import br.com.zup.beagle.android.context.operations.strategy.array.ArrayOperationTypes
+import br.com.zup.beagle.android.context.operations.strategy.comparison.ComparisionOperationTypes
+import br.com.zup.beagle.android.context.operations.strategy.invalid.InvalidOperation
+import br.com.zup.beagle.android.context.operations.strategy.number.NumberOperationTypes
+import br.com.zup.beagle.android.context.operations.strategy.string.StringOperationTypes
+import br.com.zup.beagle.android.context.operations.parameter.Parameter
+import br.com.zup.beagle.android.context.operations.parameter.ParameterTypes
+import br.com.zup.beagle.android.context.operations.parameter.removeWhiteSpaces
+
+internal class ExceptionController {
+
+    internal fun checkOperation(operation: Operation) {
+        val type = operation.operationStrategy
+
+        if (type is InvalidOperation) {
+            throw ExceptionFactory.createException(
+                exceptionTypes = type.operationType as ExceptionOperationTypes,
+                details = operation.operationToken
+            )
+        }
+        else if (operationHasIncorrectSyntax(operation)) {
+            throw ExceptionFactory.createException(
+                exceptionTypes = ExceptionOperationTypes.MISSING_DELIMITERS,
+                details = operation.operationToken
+            )
+        }
+    }
+
+    fun checkParameter(parameter: Parameter) {
+        val operationType = parameter.operation.operationStrategy?.operationType
+
+        if (parameter.arguments.isNotEmpty()) {
+            if (operationType is StringOperationTypes) {
+                checkParameterFromStringOperation(operationType, parameter)
+            } else if (operationType is NumberOperationTypes ||
+                operationType is ComparisionOperationTypes) {
+                checkParameterFromNumberOperation(parameter)
+            } else if (operationType is ArrayOperationTypes) {
+                checkParameterFromArrayOperations(operationType, parameter)
+            }
+        } else {
+            throw ExceptionFactory.createException(
+                ExceptionParameterTypes.REQUIRED_ARGS,
+                parameter.operation,
+                parameter.arguments.size.toString()
+            )
+        }
+    }
+
+    private fun checkParameterFromArrayOperations(operationType: Operations, parameter: Parameter) {
+        if ((parameter.arguments.size < 3 && operationType == ArrayOperationTypes.INSERT) ||
```

Comment:
<table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/fe9abccbff41e712f450517f44ab14629f752b4f/android/beagle/src/main/java/br/com/zup/beagle/android/context/operations/exception/ExceptionController.kt#L74"></span>This condition is too complex (5). Defined complexity threshold for conditions is set to ‘4’</td>
    </tr>
  </tbody>
</table>

Reply:
I agree with the bot, you can create some methods to separate the logic you want to verify in the operator, example:

```suggestion
        if ((isInsertAndWithCorrectSize(parameter, operation) ||
        
        fun isInsertAndWithCorrectSize(parameter, operation) -> parameter.arguments.size < 3 && operationType == ArrayOperationTypes.INSERT
```

Reply:
Done